internal/protocols/ethereum/signer.go

```diff
@@ -45,11 +49,8 @@ func (e Signer) Sign(opts signer.Options) (signedTransaction interface{}, err er
 	if opts == nil {
 		return nil, errors.New("opts must not be nil")
 	}
-	pk, ok := e.privateKey.(*secp256k1.PrivateKey)
-	if !ok {
-		return nil, errors.Errorf("invalid key type")
-	}
-	ecdsa, err := pk.ECDSA()
+
+	ecdsa, err := validatePrivateKeyType(e.privateKey)
```

Comment:
GolangCI is seeing that this name is "shadowed". In this case `ecdsa` is imported via `"crypto/ecdsa"` and you have used it as a variable name, this can cause confusion as `ecdsa` does something different before and after this line. I would suggest renaming the variable to `ecdsaPrivKey` is similar.